### PR TITLE
add alerts to psql encrypted backup job

### PR DESCRIPTION
**Pull request by ajoaugustine**
Add alerts to postgresql backup job

### CODE REVIEW

File: kube/services/jobs/psql-db-backup-encrypt-job.yaml [hunk 1]
```diff
@@ -206,6 +211,7 @@ spec:
               # Logging the failed backups
               if [ -n "$failed_backups" ]; then
                 gen3_log_info "Failed backups:\n$failed_backups"
+                curl -X POST --data-urlencode "payload={\"text\": \"*Backup failed* for psql-db-backup-encrypt on ${hostname}  Cluster: ${environment} Namespace: ${namespace} at $(date).\nFailed backups: ${failed_backups}\"}" $slackWebHook
```
Comment (jawadqur) [inline kube/services/jobs/psql-db-backup-encrypt-job.yaml:214]: Can we check if the slackwebhook is set to alert, and if it's not don't alert. Similar to other jobs in cloud-auto

File: kube/services/jobs/psql-db-backup-encrypt-job.yaml [hunk 2]
```diff
@@ -138,6 +143,18 @@ spec:
               # Import the public key
               echo "$PUBLIC_KEY" | gpg --import
 
+              # Check and retrieve Slack webhook if not provided
+              slack=false
+              if [[ "${slackWebHook}" == 'None' || -z "${slackWebHook}" ]]; then
+                slackWebHook=$(g3kubectl get configmap global -o jsonpath={.data.slack_alarm_webhook})
+              fi
```
Comment (jawadqur) [inline kube/services/jobs/psql-db-backup-encrypt-job.yaml:150]: We should not do kubectl operations from within the job itself, we should mount the secrets/configmaps needed and in this script just check if they're present.

File: kube/services/jobs/psql-db-backup-encrypt-job.yaml [hunk 3]
```diff
@@ -38,6 +38,11 @@ spec:
                   key: environment
             - name: JENKINS_HOME
               value: "devterm"
+            - name: slackWebHook
+              valueFrom:
+                configMapKeyRef:
+                  name: global
+                  key: slack_alarm_webhook
```
Comment (jawadqur) [inline kube/services/jobs/psql-db-backup-encrypt-job.yaml:45]: Can we add optional: true like here:

https://github.com/uc-cdis/cloud-automation/blob/ef9f921d93b49dafb1005ff8eda1226bd21fede4/kube/services/jobs/etl-job.yaml#L73-L78

File: kube/services/jobs/psql-db-backup-encrypt-job.yaml (hunk 2 above)
Comment (ajoaugustine) [reply kube/services/jobs/psql-db-backup-encrypt-job.yaml]: done. thanks

File: kube/services/jobs/psql-db-backup-encrypt-job.yaml (hunk 3 above)
Comment (ajoaugustine) [reply kube/services/jobs/psql-db-backup-encrypt-job.yaml:45]: added

**Review by jawadqur — APPROVED** [review]

**Pull request merged**